routes/route.js

```diff
@@ -1,101 +1,152 @@
+// route to get user details
+router.get('/get', async (req, res) => {
+  const token = req.headers.token;
+  if (!token) {
+    return res.send(
+      'Login required : Please login before accessing protected routes'
+    );
+  }
+
+  try {
+    const valid = await Access.findOne({ access_token: token }, function(err,user){
+        if(user.expired()){
+            return res.send("login expired")
+        }
+    })
+    if (!valid) {
+      return res.send('Login error : please login again');
     }
-    
-    
+    const user = await Client.findOne({ _id: valid.user_id });
+    res.send(user);
+  } catch (err) {
+      console.log(err)
+    return res.status(400).send('Invalid login');
+  }
 });
 
-router.put('/delete', async(req,res) => {
-    try{
-    const token =  req.headers.token
-    if (!token) {return res.send("Login required : Please login before accessing protected routes")} 
-
-    
-    const valid = await Client.findOne({_id : token})
-    
-    const deleted = await Client.deleteOne({username : valid.username})
-    
-    if (deleted.deleteCount == 1) { return res.send("User Deleted Successfully.")}
-    } catch (err) {
-    res.send("deletion Error")
+// route to delete request
+router.put('/delete', async (req, res) => {
+  try {
+    const token = req.headers.token;
+    if (!token) {
+      return res.send(
+        'Login required : Please login before accessing protected routes'
+      );
     }
-});
 
-router.get('/list/:page', async(req,res) => {
-    const page = parseInt(req.params.page)
-    if (page < 0) { return res.send("Invalid request")}
+    const valid = await await Access.findOne({ access_token: token }, function(err,user){
+        if(user.expired()){
+            return res.send("login expired")
+        }
+    })
+    const user = await Client.findOne({ _id: valid.user_id });
+    const deleted = await Client.deleteOne({ username: user.username });
 
-    const entries = (page +1)*10 
+    if (deleted.deleteCount == 1) {
+      return res.send('User Deleted Successfully.');
+    }
+  } catch (err) {
+    res.send('deletion Error');
+  }
+});
 
-    const data = await Client.find({});
-    
+// route to get all users
+router.get('/list/:page', async (req, res) => {
+  const page = parseInt(req.params.page);
+  if (page < 0) {
+    return res.send('Invalid request');
+  }
 
-    if (data.length < (entries - 10) ) {
-        return res.send(data)
-    } else if ((entries - 10) < data.length < entries ){
-        return res.send(data.slice((entries-10),data.length))
-    } else {
-        return res.send(data.slice((entries-10,entries)))
-    }
-    
-})
+  const entries = (page + 1) * 10;
 
+  const data = await Client.find({});
 
+  if (data.length < entries - 10) {
+    return res.send(data);
+  } else if (entries - 10 < data.length < entries) {
+    return res.send(data.slice(entries - 10, data.length));
+  } else {
+    return res.send(data.slice((entries - 10, entries)));
+  }
```

Comment:
all this manipulation can be done by database query itself.
also on large data set this code will affect the performance just because you are manipulating on all dataset.